### ISSUE TITLE
An issue about function constructor

### ISSUE THREAD

**Issue by YuanWangC** [facebook/hermes#154]
When using constructor to define a new function and let the argument be `"return this;"` ,  its output is abnormal. If I change the argument into `"return true;"` or `"print(\"hello\");return this;"`, the program will execute normally and get the expected output. I suspect it is a bug.
#### Test case:
```
var NISLFuzzingFunc = function(){
    var f = new Function("return this;");
    //var f = new Function("return true;");
    // var f = new Function("print(\"hello\");return this;");
    var res = f();
    print(res);
    print( f.toString === undefined );
    print(f);
}
NISLFuzzingFunc();
```
#### Command:
`build_release/bin/hermes -w testcase.js `
#### Output:
```
[object global]
true
TypeError: Cannot determine default value of object
```
#### Expected Output:
```
[object global]
false
function anonymous() { [bytecode] }
```

**Comment by ljharb** [comment]
The second one will be false, since Functions always start in sloppy mode, and the global object’s toString is Object.prototype.toString.

The third seems like a bug.

**Comment by tmikov** [comment]
Thank you for reporting this bug. It was caused by special handling of "return this;" for configurations where eval() is disabled. It has actually already been fixed in 4b732099486713cdaa3003b869d8fc6dc5136ab5.

The fix did reveal another related bug (the length property is invalid), which we are working on now.
